modules/entrezdirect/esummary/main.nf

```diff
@@ -0,0 +1,36 @@
+process ENTREZDIRECT_ESUMMARY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::entrez-direct=13.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/entrez-direct:13.9--pl526h375a9b1_1':
+        'quay.io/biocontainers/entrez-direct:13.9--pl526h375a9b1_1' }"
+
+    input:
+    tuple val(meta), val(database), val(uid)
+
+    output:
+    tuple val(meta), path(esummary_output), emit: xml_esummary
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    esummary_output = "${prefix}.esummary.xml.txt"
+    """
+    esummary \\
+        -db $database \\
+        -id $uid \\
+        $args \\
+        > ${esummary_output} 2> ${prefix}.esearch.log
```

Comment:
There is no value in predefining `esummary_output`. Just put `${prefix}.esummary.xml.txt` here.